### PR TITLE
Export Download gift card PDF to API + tests

### PR DESCRIPTION
**Pull request by Roshyo**

### CODE REVIEW

File: src/Api/Controller/Action/DownloadGiftCardPdfAction.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        GiftCardChannelConfigurationProviderInterface $configurationProvider,
+        GiftCardPdfGeneratorInterface $giftCardPdfGenerator
+    ) {
+        $this->configurationProvider = $configurationProvider;
+        $this->giftCardPdfGenerator = $giftCardPdfGenerator;
+    }
+
+    public function __invoke(GiftCardInterface $data): Response
+    {
+        $giftCard = $data;
```
Comment (loevgaard) [inline src/Api/Controller/Action/DownloadGiftCardPdfAction.php]:
```suggestion
```

File: src/Api/Controller/Action/DownloadGiftCardPdfAction.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    public function __construct(
+        GiftCardChannelConfigurationProviderInterface $configurationProvider,
+        GiftCardPdfGeneratorInterface $giftCardPdfGenerator
+    ) {
+        $this->configurationProvider = $configurationProvider;
+        $this->giftCardPdfGenerator = $giftCardPdfGenerator;
+    }
+
+    public function __invoke(GiftCardInterface $data): Response
```
Comment (loevgaard) [inline src/Api/Controller/Action/DownloadGiftCardPdfAction.php:28]:
```suggestion
    public function __invoke(GiftCardInterface $giftCard): Response
```

File: src/Generator/GiftCardPdfGenerator.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusGiftCardPlugin\Generator;
+
+use Knp\Bundle\SnappyBundle\Snappy\Response\PdfResponse;
+use Knp\Snappy\Pdf;
```
Comment (loevgaard) [inline src/Generator/GiftCardPdfGenerator.php]: There is no interface for this?

File: src/Resources/config/api_resources/GiftCard.xml [hunk 4]
```diff
@@ -40,6 +40,16 @@
                 </attribute>
             </itemOperation>
 
+            <itemOperation name="admin_download_pdf">
+                <attribute name="method">GET</attribute>
+                <attribute name="path">/admin/gift-cards/{code}/pdf</attribute>
+                <attribute name="controller">setono_sylius_gift_card_plugin.api.download_gift_card_pdf</attribute>
+                <attribute name="openapi_context">
+                    <attribute name="summary">Download the associated PDF.</attribute>
```
Comment (loevgaard) [inline src/Resources/config/api_resources/GiftCard.xml]:
```suggestion
                    <attribute name="summary">Download the gift card as a PDF.</attribute>
```

File: src/Resources/config/services/misc.xml [hunk 5]
```diff
@@ -3,12 +3,22 @@
...
         </service>
 
+        <service id="setono_sylius_gift_card.generator.gift_card_pdf"
+                 class="Setono\SyliusGiftCardPlugin\Generator\GiftCardPdfGenerator">
+            <argument type="service" id="twig"/>
+            <argument type="service" id="knp_snappy.pdf"/>
+        </service>
+        <service id="Setono\SyliusGiftCardPlugin\Generator\GiftCardPdfGeneratorInterface"
+                 alias="setono_sylius_gift_card.generator.gift_card_pdf" />
+        <!-- Generators -->
```
Comment (loevgaard) [inline src/Resources/config/services/misc.xml]:
```suggestion
```

**Review by loevgaard — APPROVED** [review]
A few comments

File: src/Generator/GiftCardPdfGenerator.php (hunk 3 above)
Comment (Roshyo) [reply src/Generator/GiftCardPdfGenerator.php]: I guess we can use `Knp\Snappy\GeneratorInterface` indeed

File: src/Resources/config/api_resources/GiftCard.xml (hunk 4 above)
Comment (Roshyo) [reply src/Resources/config/api_resources/GiftCard.xml]: Is it needed ? I mean, this block is already contextualized to GC: 
image

Comment (loevgaard) [reply src/Resources/config/api_resources/GiftCard.xml]: Hehe, I don't mind :D I just thought it sounded more concise

**Pull request merged**